numpyro/distributions/constraints.py

```diff
@@ -78,6 +78,14 @@ def __call__(self, value):
         return (value > self.lower_bound) & (value < self.upper_bound)
 
 
+class _Multinomial(Constraint):
+    def __init__(self, upper_bound):
+        self.upper_bound = upper_bound
+
+    def __call__(self, value):
+        return np.sum(value, -1) == self.upper_bound
```

Comment:
We miss a check for nonnegative here. And I guess you mean `total_count` instead of `upper_bound`?

Reply:
`total_count` is stored as `upper_bound`. I just wanted to keep the same convention as the rest of the constraints class in terms of using `upper_bound` and `lower_bound` as attributes, but we can change that if it looks confusing. Will add the non-negativity constraint.